addon/components/hot-replacement-component.js

```diff
@@ -61,8 +62,10 @@ const HotReplacementComponent = Ember.Component.extend(HotComponentMixin, {
   }).volatile(),
 
   __willLiveReload (event) {
-    if (matchingComponent(this.get('baseComponentName'), event.modulePath)) {
+    const baseComponentName = this.get('baseComponentName');
+    if (matchingComponent(baseComponentName, event.modulePath)) {
       event.cancel = true;
+      clearRequirejs(this, baseComponentName);
```

Comment:
This seems pretty similar to what `ember-cli-hot-loader/utils/clear-container-cache` does and currently it's called from https://github.com/toranb/ember-cli-hot-loader/blob/master/addon/components/hot-replacement-component.js#L73 which is called from https://github.com/toranb/ember-cli-hot-loader/blob/2eb0376c909607e80059360b7382ea93e31f8a11/addon/mixins/hot-component.js#L12

I think there is a bit of duplication that we could get away with, some suggestions:  
* Can we inline clear-requirejs with clear-container-cache
* There is already an event that handles `willReload` I suggest we leave the call there instead of clearing the cache when canceling `liveReload`